src/components/add-cash/ApplePayButton.js

```diff
@@ -46,27 +42,46 @@ const ApplePayButton = ({ disabled, onDisabledPress, onSubmit }) => {
       ];
   }, [colors.blueGreyDark50, colors.shadow, disabled, darkMode]);
 
-  const disabledAnimation = useTimingTransition(!disabled, {
-    duration: 66,
-    ease: EasingNode.out(EasingNode.ease),
-  });
+  const disabledAnimation = useSharedValue(disabled ? 0 : 1);
 
-  const backgroundColor = mixColor(
-    disabledAnimation,
-    darkMode
-      ? colors.alpha(darkModeThemeColors.grey20, 0.3)
-      : lightModeThemeColors.blueGreyDark50,
-    darkMode ? darkModeThemeColors.darkModeDark : lightModeThemeColors.dark
-  );
+  useLayoutEffect(() => {
+    disabledAnimation.value = withTiming(disabled ? 0 : 1, {
+      duration: 66,
+      easing: Easing.out(Easing.ease),
+    });
+  }, [disabled, disabledAnimation]);
 
-  const defaultShadowOpacity = interpolate(disabledAnimation, {
-    inputRange: [0, 1],
-    outputRange: [0, 1],
+  const animatedProps = useAnimatedProps(() => {
+    const outputColors = darkMode
+      ? [
+          colors.alpha(darkModeThemeColors.grey20, 0.3),
+          darkModeThemeColors.darkModeDark,
```

Comment:
could you check if it actually adds the whole theme to the context of worklet or just these colors?

Reply:
I went ahead and just removed it and placed it in a component